importer/helpers/dagbuilder.go

```diff
@@ -237,3 +345,104 @@ func (db *DagBuilderHelper) Maxlinks() int {
 func (db *DagBuilderHelper) Close() error {
 	return db.batch.Commit()
 }
+
+// AddNodeAndClose adds the last `ipld.Node` from the DAG and
+// closes the builder. It returns the same `node` passed as
+// argument.
+func (db *DagBuilderHelper) AddNodeAndClose(node ipld.Node) (ipld.Node, error) {
+	err := db.batch.Add(node)
+	if err != nil {
+		return nil, err
+	}
+
+	err = db.Close()
+	if err != nil {
+		return nil, err
+	}
+
+	return node, nil
+}
+
+// FSNodeOverDag encapsulates an `ft.FSNode` that will be stored in a
+// `dag.ProtoNode`. Instead of just having a single `ipld.Node` that
+// would need to be constantly (un)packed to access and modify its
+// internal `FSNode` in the process of creating a UnixFS DAG, this
+// structure stores an `FSNode` cache to manipulate it (add child nodes)
+// directly , and only when the node has reached its final (immutable)
+// state it is committed to a single (indivisible) `ipld.Node`.
```

Comment:
“state it is committed” → “state is it committed”

Reply:
I might add that moving to its final state is done by calling `Commit`:

> …and only when the node has reached its final state (signaled by calling `Commit()`) is it committed…

Otherwise it sounds like that happens automatically as a result of doing some other routine action.